### PR TITLE
Allow 401-Headers 

### PR DESCRIPTION
**Pull request by heiglandreas**
This PR allows 401-Headers for actions that require authentication. 

These actions are currently returning a 400-Status which I'd consider too generic.

### CODE REVIEW

File: src/controllers/EventsController.php [hunk 1]
```diff
@@ -539,6 +539,7 @@ function ($tag) {
      */
     public function createTrack($request, $db)
     {
+        // Should we not check for a login?
```
Comment (heiglandreas) [inline src/controllers/EventsController.php]: And fail fast

File: src/controllers/TracksController.php [hunk 2]
```diff
@@ -25,6 +25,7 @@ public function getAction($request, $db)
 
     public function editTrack($request, $db)
     {
+        // Should we not check for login?
```
Comment (heiglandreas) [inline src/controllers/TracksController.php]: And fail fast

File: src/controllers/TracksController.php [hunk 3]
```diff
@@ -74,6 +75,7 @@ public function editTrack($request, $db)
 
     public function deleteTrack($request, $db)
     {
+        // Should we not check for login?
```
Comment (heiglandreas) [inline src/controllers/TracksController.php]: And fail fast

**Comment by Jean85** [comment]
I agree with the fail-fast approach! LGTM! :+1:

**Comment by liam-wiltshire** [comment]
@heiglandreas yes, we should be checking for login on tracks - it's not critical in the sense of it being a security hole, as we are checking that the user can edit that event, but equally we could advise some unnecessary  DB reads by doing so.

Will raise a ticket in JIRA to add that in - would be a good one for the hackathon.

Just testing the other changes now, but I can't see any issues so far, so all being well will merge later this eve.

**Pull request merged**